api/src/main/java/org/openmrs/Condition.java

```diff
@@ -310,7 +310,9 @@ public boolean equals(Object o) {
 		if (verificationStatus != condition.verificationStatus) {
 			return false;
 		}
-		if (!this.condition.getCoded().equals(condition.getCondition().getCoded())) {
+		if (this.condition.getCoded() != null ?
```

Comment:
I think you have made this a bit more complicated. Why not just add a this.condition.getCoded() != null to the original if condition?

Reply:
Okay. Let me rectify that.